### ISSUE TITLE
EventHubTrigger does not trigger when using .NET Core 2.0

### ISSUE THREAD

**Issue by symlynk** [Azure/azure-webjobs-sdk#1484]
The following code works when using a console app using .NET Framework 4.6.1, but when using .NET Core, the method 'Handle' with the EventHubTrigger is never called.

```
    class Program
    {
        static void Main( string[] args )
        {
            var config = new JobHostConfiguration();

            config.DashboardConnectionString = "..."; // replace ... with storage account connection string
            config.StorageConnectionString = "..."; // replace ... with storage account connection string

            var eventHubConfig = new EventHubConfiguration();

            eventHubConfig.AddReceiver( "example", "..." ); // replace ... with event hub connection string

            config.UseEventHub( eventHubConfig );

            var host = new JobHost( config );

            host.RunAndBlock();
        }
    }

    public class EventHandler
    {
        public void Handle( [EventHubTrigger( "example" )] EventData data )
        {
        }
    }
```

As per issue #1483 "Exception thrown: 'Microsoft.WindowsAzure.Storage.StorageException' in System.Private.CoreLib.dll" is written to the output window continuously.

Microsoft.Azure.EventHubs (1.1.0)
Microsoft.Azure.WebJobs.Extensions.EventHubs (3.0.0-beta4)
Microsoft.Azure.WebJobs.ServiceBus (3.0.0-beta4)

**Comment by EhRom** [comment]
I tested this code only with Microsoft.Azure.WebJobs.Extensions.EventHubs (3.0.0-beta4).
It worked. 

May it comes from a conflict between these pacakges ?

**Comment by paulbatum** [comment]
This issue might need to be moved to https://github.com/Azure/azure-webjobs-sdk-extensions but I'll wait for a first pass investigation to confirm this.

**Comment by evgenisokolov** [comment]
Have completely the same behavior. Tried also Microsoft.Azure.WebJobs.Extensions.EventHubs (3.0.0-beta5) but have the same exception in output and job host even not starting

**Comment by paulbatum** [comment]
I was able to reproduce this and I think I know what is going on.

**TL;DR  - to get working bits as of today, you need to use the right versions of these packages:**
- Microsoft.Azure.WebJobs.Extensions.EventHubs (3.0.0-beta5)
- Microsoft.Azure.EventHubs (1.0.3)

In order to be able to use a newer version of `Microsoft.Azure.EventHubs`, you'll need to wait for us to do an update of the extension. We need to adapt to breaking changes made in the latest release of this library. This is tracked here: https://github.com/Azure/azure-webjobs-sdk/issues/1710

### Details

As far as I can tell, version 1.1.0 of `Microsoft.Azure.EventHubs` violated semver. You can see this if you install `Microsoft.Azure.EventHubs 1.1.0` + `Microsoft.Azure.WebJobs.Extensions.EventHubs (3.0.0-beta5)`:

```
Error processing EventHub message from Partition Id:0, Owner:a28cf0c0-a16a-40d5-943d-1192822242b2, EventHubPath:hublocal2
System.MissingMethodException: Method not found: 'Microsoft.Azure.EventHubs.PartitionReceiver Microsoft.Azure.EventHubs.EventHubClient.CreateEpochReceiver(System.String, System.String, System.DateTime, Int64)'.
   at Microsoft.Azure.EventHubs.Processor.EventHubPartitionPump.<OpenClientsAsync>d__5.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.EventHubs.Processor.EventHubPartitionPump.OpenClientsAsync()
   at Microsoft.Azure.EventHubs.Processor.EventHubPartitionPump.<OnOpenAsync>d__4.MoveNext()
```
Note that the call to `CreateEpochReceiver` comes from `Microsoft.Azure.EventHubs.Processor.EventHubPartitionPump`. So this is a break that occurred between two Event Hubs packages.

When you reference `Microsoft.Azure.EventHubs 2.0.0`, breaking changes made in the major release of the event hubs SDK combined with a [bug in webjobs error handling](https://github.com/Azure/azure-webjobs-sdk/issues/1703) result in a webjob that simply does not run. With that bug being fixed, I expect the behavior will change to an appropriate error message being logged.

**Comment by paulbatum** [comment]
I am closing this as answered. See my above answer for links to the relevant issues.
